### PR TITLE
Command history

### LINKED ISSUE #129: Implement command history
By using the command history, users can use the up and down key's to scroll through previously typed commands. 
They can then press [Enter] to execute them.

### PR DESCRIPTION
**Pull request by JinHao-L**
Implement command history
* Users can now use the :arrow_up: and :arrow_down: button to traverse through the command history
* Update Logic UML diagram

closes #129

### CODE REVIEW

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 1]
```diff
@@ -17,18 +20,75 @@
...
+        this.commandHistory = commandHistory;
+        this.historyPointer = commandHistory.size();
+    }
+
+    private boolean hasPreviousHistory() {
+        return historyPointer > 0;
+    }
+
+    /**
+     * Get previous command input.
```
Comment (xz0127) [inline src/main/java/seedu/address/ui/CommandBox.java]: Perhaps it should be `Gets` here?

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 2]
```diff
@@ -17,18 +20,75 @@
...
+            setInput(commandHistory.get(historyPointer));
+        }
+    }
+
+    private boolean hasNextHistory() {
+        return historyPointer < commandHistory.size() - 1;
+    }
+
+    /**
+     * Get next command input.
```
Comment (xz0127) [inline src/main/java/seedu/address/ui/CommandBox.java]: Perhaps it should be `Gets` here?

**Review by xz0127 — APPROVED** [review]
Other than the minor comments mentioned, the rest LGTM! 👍

File: src/main/java/seedu/address/ui/CommandBox.java (hunk 1 above)
Comment (JinHao-L) [reply src/main/java/seedu/address/ui/CommandBox.java]: Thanks for spotting it!

File: src/main/java/seedu/address/ui/CommandBox.java (hunk 2 above)
Comment (JinHao-L) [reply src/main/java/seedu/address/ui/CommandBox.java]: Thanks for spotting it!

**Pull request merged**